samples/pipeline_v1alpha1_task.yaml

```diff
@@ -5,16 +5,18 @@ metadata:
   namespace: default
 spec:
     inputs:
-        sources:
-            - name: workspace
+        resources:
+           - resourceRef:
+               name: resource-name
+             name: workspace
         params:
-            - name: PATH_TO_DOCKERFILE
-              type: string
+           - name: PATH_TO_DOCKERFILE
+             value: string
     outputs:
-        artifacts:
-            - name: builtImage
-              type: image
-              storeKey: registry # registy is the name of the ArtifactStore
+        resources:
+         - resourceRef:
+             name: registry
+           name: builtImage 
```

Comment:
I'm confused why we need a `resourceRef` here and what it would be referencing

We're hooking up all the refs in the `Pipeline`, and the `Task` doesn't know what `Resource` it will be running against until it is supplied.

Can you explain more about why this is needed here? (maybe it isn't?)